### PR TITLE
Add settings support for duplicate currency codes

### PR DESCRIPTION
**Pull request by peachbits**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201712843019691

### CODE REVIEW

File: src/selectors/WalletSelectors.js [hunk 1]
```diff
@@ -73,44 +80,36 @@ export const convertCurrencyFromExchangeRates = (
   return convertedAmount
```
Comment (Jon-edge) [inline src/selectors/WalletSelectors.js:80]: Combine convertCurrencyWithoutState and convertCurrencyFromExchangeRates

**Review by Jon-edge — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -95,7 +95,7 @@
     "@react-native-firebase/messaging": "^10.0.0",
     "async-lock": "^1.1.2",
     "biggystring": "^3.0.2",
-    "cleaners": "^0.3.7",
+    "cleaners": "^0.3.12",
```
[the remark this answers is not in the document]
Comment (peachbits) [reply package.json:98]: This is so I could use asValue. The library is already included but for a different dep so the corresponding yarn.lock change isn't substantial

File: src/components/scenes/SendScene.js [hunk 3]
```diff
@@ -442,14 +446,17 @@ class SendComponent extends React.PureComponent<Props, State> {
...
     if (recipientAddress) {
-      const transactionFee = convertTransactionFeeToDisplayFee(wallet, selectedCurrencyCode, exchangeRates, transaction, settings)
-      const fiatAmount = transactionFee.fiatAmount === '0' ? '0' : ` ${transactionFee.fiatAmount}`
-      const feeSyntax = `${transactionFee.cryptoSymbol ?? ''} ${transactionFee.cryptoAmount} (${transactionFee.fiatSymbol ?? ''}${fiatAmount})`
+      const wallet = currencyWallets[selectedWalletId]
+      const feeDisplayDenomination = getDisplayDenomination(wallet.currencyInfo.pluginId, wallet.currencyInfo.currencyCode)
+      const feeDefaultDenomination = getExchangeDenomination(wallet.currencyInfo.pluginId, wallet.currencyInfo.currencyCode)
+      const transactionFee = convertTransactionFeeToDisplayFee(wallet, exchangeRates, transaction, feeDisplayDenomination, feeDefaultDenomination)
+
+      const feeSyntax = `${transactionFee.cryptoSymbol || ''} ${transactionFee.cryptoAmount} (${transactionFee.fiatSymbol || ''} ${transactionFee.fiatAmount})`
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/components/scenes/SendScene.js:459]: Good catch definitely shouldn't have been

File: src/components/navigation/TransactionDropdown.js [hunk 4]
```diff
@@ -70,13 +70,18 @@ const ConnectedTransactionDropdown = connect<StateProps, DispatchProps, OwnProps
...
+    if (wallet != null) {
+      const displayDenomination = getDisplayDenomination(state, wallet.currencyInfo.pluginId, currencyCode)
+      const { symbol, name, multiplier } = displayDenomination
+      const displayAmount = convertNativeToDisplay(multiplier)(nativeAmount)
+      return {
+        message: sprintf(s.strings.bitcoin_received, `${symbol ? symbol + ' ' : ''}${displayAmount} ${name}`)
+      }
+    } else {
+      return {
+        message: sprintf(s.strings.bitcoin_received, '')
```
[the remark this answers is not in the document]
Comment (peachbits) [reply src/components/navigation/TransactionDropdown.js:83]: Good call, fixed

**Review by Jon-edge — APPROVED** [review]

**Pull request merged**